dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -13,6 +13,14 @@ class ApplicationsController < ::ApplicationController
     REGIONAL_PARTNERS_ALL = "all"
     REGIONAL_PARTNERS_NONE = "none"
 
+    TYPES_BY_ROLE = {
+      csd_teachers: TEACHER_APPLICATION_CLASS,
+      csp_teachers: TEACHER_APPLICATION_CLASS,
+      csa_teachers: TEACHER_APPLICATION_CLASS
+    }
+    ROLES = TYPES_BY_ROLE.keys
+    TEACHER_ROLES = ROLES.select {|role| role.to_s.include?('teachers')}
```

Comment:
This new rule does reveal a few places in the codebase where we are declaring constants or in some cases invoking setup methods like `prepend_before_action` down toward the bottom of the file, in amongst the private methods. I was not intending to catch such cases with this rule (going in, I was only thinking about the relative ordering of method definitions), but after seeing it in practice I think I like it.

In general, I think the best place for constants to be declared and for modifier methods to be invoked is right at the top of the file. This may lead to some situations in which some constants are far removed from the methods in which they are relevant, but the Ruby way to deal with that is to extract functionality into modules and/or concerns whenever possible, rather than to rely on manual ordering within a single large class definition.